salt/grains/metadata.py

```diff
@@ -129,6 +56,8 @@ def _search(prefix="latest/"):
                 ret[line] = retdata
     return salt.utils.data.decode(ret)
 
-
-def metadata():
-    return _search()
+def main():
```

Comment:
`main()`?

```suggestion
def metadata():
```

I guess your issue is that you import `import salt.utils.aws as metadata`, don't.
Instead do:

```python
import salt.utils.aws

...  

salt.utils.aws.get_metadata()
```

Or, 
```
import salt.utils.aws as aws
...  
aws.get_metadata()
```